### PR TITLE
Initial cloud FP doc

### PR DESCRIPTION
**Pull request by jfdenise**
An initial documentation that covers launch scripts and Galleon layers.

### CODE REVIEW

**Comment by jfdenise** [comment]
@jmesnil , bootstraping the README.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,7 @@
 # eap-cloud-galleon-pack
 Galleon feature-pack for EAP on the Cloud
```
Comment (jmesnil) [inline README.md:2]: Do we want to say "EAP 8" instead of "EAP"?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,7 @@
...
+
+The ``org.jboss.eap.cloud:eap-cloud-galleon-pack`` Galleon feature-pack provisions a set of additional features allowing you to configure an \
+EAP server to work on the cloud.
```
Comment (jmesnil) [inline README.md:5]: s/to work/to run ?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,7 @@
 # eap-cloud-galleon-pack
 Galleon feature-pack for EAP on the Cloud
+
+The ``org.jboss.eap.cloud:eap-cloud-galleon-pack`` Galleon feature-pack provisions a set of additional features allowing you to configure an \
+EAP server to work on the cloud. 
+
+[Detailed documentation](doc/index.md) of the EAP cloud Galleon feature-pack.
```
Comment (jmesnil) [inline README.md:7]: That's a mouthful but we should use the same terminology to be consistent "Galleon feature-pack for EAP on the Cloud" (or say we will refer to it as the "EAP cloud feature pack"?)

File: doc/index.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Building an EAP server for the cloud
```
Comment (jmesnil) [inline doc/index.md:1]: s/cloud/Cloud

File: doc/index.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+In order to provision an EAP server for the cloud, you have first to integrate the [EAP Maven plugin](https://github.com/jbossas/eap-maven-plugin/)
```
Comment (jmesnil) [inline doc/index.md:3]: idem

File: doc/index.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+into the pom.xml file of your project.
+
+The "package" goal of the plugin allows you to:
+
+* Provision an EAP server tailored to your needs.
+* Incorporate custom content (keystores, properties files, ...).
+* Execute CLI scripts.
+* Deploy your application.
```
Comment (jmesnil) [inline doc/index.md:11]: Deploy your application in the provisioned server

File: doc/index.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Building an EAP server for the cloud
+
+In order to provision an EAP server for the cloud, you have first to integrate the [EAP Maven plugin](https://github.com/jbossas/eap-maven-plugin/)
+into the pom.xml file of your project.
+
+The "package" goal of the plugin allows you to:
+
+* Provision an EAP server tailored to your needs.
+* Incorporate custom content (keystores, properties files, ...).
+* Execute CLI scripts.
```
Comment (jmesnil) [inline doc/index.md:10]: Execute CLI scripts to configure the EAP server

File: doc/index.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+For example:
+
+```xml
+<feature-packs>
+  <feature-pack>
+    <location>org.jboss.eap:wildfly-ee-galleon-pack</location>
+  </feature-pack>
+  <feature-pack>
+    <location>org.jboss.eap.cloud:eap-cloud-galleon-pack</location>
```
Comment (jmesnil) [inline doc/index.md:28]: names... for my own understanding and for future references, we will have 2 duos of feature packs:

* EAP 8
  * org.jboss.eap:wildfly-ee-galleon-pack
  * org.jboss.eap.cloud:eap-cloud-galleon-pack
* EAP XP
  * org.jboss.eap:wildfly-galleon-pack
  * org.jboss.eap.cloud:eap-xp-cloud-galleon-pack

Is that correct?

File: doc/index.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  </feature-pack>
+</feature-packs>
+```
+
+# Specifics of the wildfly-cloud-galleon-pack
```
Comment (jmesnil) [inline doc/index.md:33]: s/wildfly-cloud-galleon-pack/eap-cloud-galleon-pack

File: doc/launch.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
+# Server startup
+
+When an EAP server provisioned with the cloud feature-pack is started, a set of bash scripts are executed in order to adjust 
+the server configuration. These bash scripts are controlled by a set of environment variables that this document describes.
+
+When running an EAP server inside the EAP S2i runtime and builder images, you can use this [these environment variables](https://github.com/jboss-container-images/openjdk/blob/develop/modules/jvm/api/module.yaml) to configure the Java VM.
```
Comment (jmesnil) [inline doc/launch.md:6]: s/S2i/S2I (Source-to-Image)

Comment (jmesnil) [inline doc/launch.md:6]: s/you can use this/you can use

File: doc/launch.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+EAP S2i runtime and builder images are exposing a set of [environment variables](https://github.com/wildfly/wildfly-cekit-modules/blob/main/jboss/container/wildfly/run/api/module.yaml) to fine tune the server execution.
```
Comment (jmesnil) [inline doc/launch.md:8]: s/S2i/S2I

File: doc/launch.md [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+
+# Datasources
+
+Configure datasources subsystem.
```
Comment (jmesnil) [inline doc/launch.md:12]: maybe later drop a note about the relation with the `eap-datasources-feature-pack`.

File: doc/launch.md [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+
+## Environment variables
+
+[Environment variables.](https://github.com/wildfly/wildfly-cekit-modules/blob/main/jboss/container/wildfly/launch/datasources/module.yaml)
```
Comment (jmesnil) [inline doc/launch.md:16]: Note related to your PR but we need to create an index of all these environment variables...

**Review by jmesnil — CHANGES_REQUESTED** [review]
looks good to me (except a few typo).

I also wonder if we should use asciidoc to write this doc as it offers a richer way to write it.

File: README.md (hunk 1 above)
Comment (jfdenise) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (jfdenise) [reply README.md]: fixed.

File: doc/index.md (hunk 4 above)
Comment (jfdenise) [reply doc/index.md]: Fixed

File: doc/index.md (hunk 5 above)
Comment (jfdenise) [reply doc/index.md]: Fixed

File: doc/launch.md (hunk 10 above)
Comment (jfdenise) [reply doc/launch.md]: Fixed

Comment (jfdenise) [reply doc/launch.md]: Fixed

File: doc/launch.md (hunk 11 above)
Comment (jfdenise) [reply doc/launch.md]: Fixed

File: doc/launch.md (hunk 12 above)
Comment (jfdenise) [reply doc/launch.md:12]: Added a note.

File: README.md (hunk 3 above)
Comment (jfdenise) [reply README.md]: Fixed.

File: doc/index.md (hunk 6 above)
Comment (jfdenise) [reply doc/index.md]: Fixed.

File: doc/index.md (hunk 8 above)
Comment (jfdenise) [reply doc/index.md:28]: Correct.

File: doc/index.md (hunk 9 above)
Comment (jfdenise) [reply doc/index.md]: Fixed

**Review by jmesnil — APPROVED** [review]

**Pull request merged**